dandi/files/bases.py

```diff
@@ -185,7 +199,23 @@ def get_validation_errors(
                     e,
                     extra={"validating": True},
                 )
-                return [str(e)]
+                # TODO: how do we get **all** errors from validation - there must be a way
+                return [
+                    ValidationResult(
```

Comment:
create a helper `pydantic_errors_to_validation_results(errors: list[dict]) -> list[ValidationResult]` which you would pass `e.errors` here and in the other location above on validation of the dandiset level metadata.

Each error seems to be a dict which looks like

```
*(Pdb) pprint(e.errors()[-1])
{'loc': ('version',), 'msg': 'field required', 'type': 'value_error.missing'}
*(Pdb) pprint(e.errors())
[{'ctx': {'pattern': '^(dandi|DANDI):\\d{6}(/(draft|\\d+\\.\\d+\\.\\d+))$'},
  'loc': ('id',),
  'msg': 'string does not match regex '
         '"^(dandi|DANDI):\\d{6}(/(draft|\\d+\\.\\d+\\.\\d+))$"',
  'type': 'value_error.str.regex'},
 {'loc': ('name',), 'msg': 'field required', 'type': 'value_error.missing'},
 {'loc': ('description',),
  'msg': 'field required',
  'type': 'value_error.missing'},
 {'loc': ('contributor',),
  'msg': 'field required',
  'type': 'value_error.missing'},
 {'loc': ('license',), 'msg': 'field required', 'type': 'value_error.missing'},
 {'loc': ('identifier',),
  'msg': 'field required',
  'type': 'value_error.missing'},
 {'loc': ('citation',), 'msg': 'field required', 'type': 'value_error.missing'},
 {'loc': ('assetsSummary',),
  'msg': 'field required',
  'type': 'value_error.missing'},
 {'loc': ('manifestLocation',),
  'msg': 'field required',
  'type': 'value_error.missing'},
 {'loc': ('version',), 'msg': 'field required', 'type': 'value_error.missing'}]

```

could look like following for the `id`

```
(Pdb) dict(id=':'.join(filter(bool, ("dandischema", e0.get("type", "UNKNOWN"), '+'.join(e0.get("loc", []))))))
{'id': 'dandischema:value_error.str.regex:id'}
(Pdb) 
{'id': 'dandischema:value_error.str.regex:id'}
```

for the message I guess just take `msg` if it is there.